tests/cases/conformance/types/nonPrimitive/nonPrimitiveAndTypeVariables.ts

```diff
@@ -0,0 +1,14 @@
+// @strict: true
+
+// Repros from #23800
+
+type A<T, V> = { [P in keyof T]: T[P] extends V ? 1 : 0; };
+type B<T, V> = { [P in keyof T]: T[P] extends V | object ? 1 : 0; };
+
+type a = A<{ a: 0 | 1 }, 0>; // { a: 0; }

```

Comment:
the baselines don't actually show that `type a = { a: 0 }` (or that `type b = { a: 0 }`.